### ISSUE TITLE
Opening multiple channels from a single ssh client throws OpenFailException

### ISSUE THREAD

**Issue by yonatankarimish** [hierynomus/sshj#612]
(note: This is the issue that made me discover issue #611 )

I've been using the library for quite some time (version 0.27.0) for opening connections to network devices and executing commands on them.  I recently encountered a problem creating new sessions from a single ssh client.

Minimal reproducible environment:
```
Ubuntu 16.04.5 LTS (GNU/Linux 4.4.0-131-generic x86_64)
JDK 11
```

Source code:
```
            for(int i = 0; i<=100; i++) {
                AtomicInteger counter = new AtomicInteger(i);
                threadingManager.submit(() -> { // submits the lambda to a central thread pool for concurrent execution
                    logger.fatal("Before starting session #"+counter.get());
                    try {
                        sshClient.startSession();
                        Thread.sleep(5000);
                        sshClient.close();
                    } catch (Exception e) {
                        e.printStackTrace();
                    }
                    logger.fatal("After closing session #"+counter.get());
                });
            }
```

Output: 
```
net.schmizz.sshj.connection.channel.OpenFailException: Opening `session` channel failed: open failed
	at net.schmizz.sshj.connection.channel.direct.AbstractDirectChannel.gotOpenFailure(AbstractDirectChannel.java:74)
	at net.schmizz.sshj.connection.channel.direct.AbstractDirectChannel.gotUnknown(AbstractDirectChannel.java:99)
	at net.schmizz.sshj.connection.channel.AbstractChannel.handle(AbstractChannel.java:204)
	at net.schmizz.sshj.connection.ConnectionImpl.handle(ConnectionImpl.java:130)
	at net.schmizz.sshj.transport.TransportImpl.handle(TransportImpl.java:500)
	at net.schmizz.sshj.transport.Decoder.decodeMte(Decoder.java:159)
	at net.schmizz.sshj.transport.Decoder.decode(Decoder.java:79)
	at net.schmizz.sshj.transport.Decoder.received(Decoder.java:231)
	at net.schmizz.sshj.transport.Reader.run(Reader.java:59)
```

I tried to run the code again, in a sequential manner:
```
            for(int i = 0; i<=100; i++) {
                logger.fatal("Before starting session #"+i);
                try {
                    sshClient.startSession();
                    Thread.sleep(200);
                    sshClient.close();
                } catch (Exception e) {
                    logger.fatal(e);
                    e.printStackTrace();
                }
                logger.fatal("After closing session #"+i);
            }
```

The 2nd output was:
```
java.lang.IllegalStateException: Not connected
	at net.schmizz.sshj.SSHClient.checkConnected(SSHClient.java:799)
	at net.schmizz.sshj.SSHClient.startSession(SSHClient.java:737)
```

Basically i'm stuck with the two issues I opened, and am unable to use the library for more than a single connection to a single device at any given time.

Thanks for your help on this matter.

**Comment by hierynomus** [comment]
`SshClient` is not reusable, see the JavaDoc.
